source/core/TDCodecVia.cpp

```diff
@@ -1834,33 +1835,33 @@ void TDViaFormatter::FormatIEEE754(TypeRef type, void* pData)
     Double value = ReadDoubleFromMemory(type, pData);
     Int32 len = 0;
     Boolean suppressInfNaN = _options._fmt.SuppressInfNaN();
+    Boolean quotedInfNaN = _options._fmt.QuotedNameInfNaN();
+
     if (isnan(value)) {
-#if 0
-        // TODO unit tests are getting different -NaNs in different cases.
-        if (signbit(value)) {
-            pBuff = "-nan";
-            len = 4;
-        } else {
-            pBuff = "nan";
-            len = 3;
-        }
-#else
         if (!suppressInfNaN) {
-            pBuff = _options._fmt.LongNameInfNaN() ?  "NaN" : "nan";
+            pBuff = _options._fmt.LongNameInfNaN() ?  "\"NaN\"" : "\"nan\"";
             len = 3;
+            if (quotedInfNaN)
```

Comment:
I wonder if there is a way to produce cleaner code. Is it costly to add the quotes to pBuff later? Increasing len in one section of the if and then increasing the pointer to Buff in the else, required more attention to understand.